backup/moodle2/restore_local_annoto_plugin.class.php

```diff
@@ -0,0 +1,100 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Annoto backup class.
+ *
+ * @package    local_annoto
+ * @subpackage annoto
+ * @copyright  Annoto Ltd.
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class restore_local_annoto_plugin extends restore_plugin {
+    /**
+     * Define the paths to be handled by the plugin in the xml dom.
+     *
+     * @return array Array of restore_path_element objects
+     */
+    protected function define_module_plugin_structure() {
+        // Path to the annoto completion data in the backup xml.
+        $paths[] = new restore_path_element('local_annoto_completion', '/module/local_annoto_completion');
+        $userinfo = $this->get_setting_value('userinfo');
+        if ($userinfo) {
+            $paths[] = new restore_path_element(
+                'local_annoto_completiondata',
+                '/module/local_annoto_completion/completiondataset/local_annoto_completiondata'
+            );
+        }
+
+        return $paths;
+    }
+
+    /**
+     * Process the completion from xml backup.
+     * Will not be called if xml element not found.
+     *
+     * @param object $data
+     */
+    public function process_local_annoto_completion($data) {
+        global $DB, $USER;
+
+        // Cast to object.
+        $data = (object)$data;
+        $userinfo = $this->get_setting_value('userinfo');
+        // Used /moodle/mod/quiz/accessrule/seb/backup/moodle2/backup_quizaccess_seb_subplugin.class.php.
+        // As a reference for case when userdata is false to use user who perform action as a modifier.
+        if (!$userinfo) {
```

Comment:
according to the quiz reference, this is not needed, $USER->id is always used

Reply:
It was added as an optional value in case when moderator want to save referred user. Deleted.